File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.*;

class Solution {
    public String solution(String[] survey, int[] choices) {
        String answer = "";
        
        Map<String, Integer> map = new HashMap<>();
        
        StringBuilder sb = new StringBuilder();
        
        map.put("R",0);
        map.put("T",0);
        map.put("C",0);
        map.put("F",0);
        map.put("J",0);
        map.put("M",0);
        map.put("A",0);
        map.put("N",0);
        
        for(int i = 0; i<survey.length;i++){
            String disagree = survey[i].substring(0,1);
            String agree = survey[i].substring(1,2);
            
            if(choices[i]<4){
                map.put(disagree, map.get(disagree)+4-choices[i]); 
            }
            if(choices[i]>4){
                 map.put(agree, map.get(agree)+choices[i]-4);
            }
          
            // System.out.println(disagree+" "+ agree);
        }
        // for(String key : map.keySet()){
        // Integer value = map.get(key);
        // System.out.println(String.format("키 : "+key+", 값 : "+value));
        // }
        
        
        if(map.get("R")>=map.get("T")) sb.append("R");
        else sb.append("T");
        
        if(map.get("C")>=map.get("F")) sb.append("C");
        else sb.append("F");
        
        if(map.get("J")>=map.get("M")) sb.append("J");
        else sb.append("M");
        
        if(map.get("A")>=map.get("N")) sb.append("A");
        else sb.append("N");
        
        answer = sb.toString();
        // System.out.println(sb.toString());
        
        return answer;
    }
}

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {

    let map = new Map();
    
    map.set('R',0);
    map.set('T',0);
    map.set('C',0);
    map.set('F',0);
    map.set('J',0);
    map.set('M',0);
    map.set('A',0);
    map.set('N',0);
    
    for(let i = 0; i<survey.length; i++){
        const type1 = survey[i].charAt(0);
        const type2 = survey[i].charAt(1);
        
        let score = choices[i];
        if(score>=4) {
            score = score-4;
            map.set(type2,map.get(type2)+score);
        }
        else{
            score = 4-score;
            map.set(type1,map.get(type1)+score);
        }
    }
    console.log(map);
    let answer = "";
    if(map.get('R')>=map.get('T')) answer+="R"
    else answer+="T"
    if(map.get('C')>=map.get('F')) answer+="C"
    else answer+="F"
    if(map.get('J')>=map.get('M')) answer+="J"
    else answer+="M"
    if(map.get('A')>=map.get('N')) answer+="A"
    else answer+="N"

    
    return answer;
}